engine/uState.ml

```diff
@@ -123,8 +123,9 @@ let context uctx =
 
 let univ_entry ~poly uctx =
   let open Entries in
-  if poly then Polymorphic_entry (context uctx)
-  else Monomorphic_entry (context_set uctx)
+  let (binders, _) = uctx.names in
+  (if poly then Polymorphic_entry (context uctx)
+   else Monomorphic_entry (context_set uctx)), binders
```

Comment:
This is not very pretty, instead do `let entry = if poly then ... in entry, binders`

Reply:
Done